include/arch/gdbstub.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_GDBSTUB_H_
+#define ZEPHYR_INCLUDE_ARCH_GDBSTUB_H_
+
+#include <inttypes.h>
+
+#if defined(CONFIG_X86)
+#include <arch/x86/ia32/gdbstub.h>
+#endif
+
+/**
+ * @brief Architecture layer debug start
+ *
+ * This function is called by @c gdb_init()
+ */
+void arch_gdb_init(void);
```

Comment:
why are these still here? needs to be in either include/sys/arch_interface.h or kernel/include/kernel_arch_interface.h

Reply:
the whole point of these headers is that we don't have all these arch interface definitions scattered all over the include space/, they are in one of two possible locations

Reply:
sorry maybe this comment was lost, I am confuse because this:

can you be a little bit more specific ? are you proposing just move the API declaration to include/sys/arch_interface.h and keep the header include/arch/gdbstub.h that will be responsible to include architecture specific header include/arch/x86/ia32/gdbtub.h ? or get rid of include/sys/arch_interface.h and include include/arch/x86/ia32/gdbstub.h directly in include/sys/arch_interface.h ? I'm ok with the move, it is just looking a little bit an alien there though.

Reply:
> are you proposing just move the API declaration to include/sys/arch_interface.h

yes.
I do not want any arch_ API prototypes/documentation other than in the two headers they currently reside in.

Need a docstring for ARCH_GDB_NUM_REGISTERS, everything that is required to be defined or implemented must have doxygen for benefit of people porting to new arches.

> keep the header include/arch/gdbstub.h that will be responsible to include architecture specific header include/arch/x86/ia32/gdbtub.h

Can't you just pull this in from x86's arch.h like the rest of the platform definitions?

Reply:
To be crystal clear: I absolutely care about making sure that the places where arch_ APIs are defined do not proliferate beyond the two headers they are already in, the only reason why it's currently two and not one is because of inline function needs.

Reply:
other than this issue and a request to open a GH enhancement ticket for CI-based testing (which can be assigned to someone else) this is a +1